### PR TITLE
ci: convert e2e to matrix

### PR DESCRIPTION
**Pull request by gartnera**
# Description

Use a matrix for e2e tests. Will port all advanced tests over here eventually.

Also enable you to request additional testing by adding labels on the PR.

Eventually this will also run on mergequeue.

Before merge, the required status checks will need to be updated.

Related to #2134 

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [x] Tested via GitHub Actions

### CODE REVIEW

**Review by lumtis — APPROVED** [review]
LGTM
We can run tests depending on the tag?

**Comment by gartnera** [comment]
> We can run tests depending on the tag?

Yep just apply the desired `*_TESTS` labels.

File: contrib/localnet/docker-compose.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ services:
     environment:
       - HOTKEY_BACKEND=file
       - HOTKEY_PASSWORD=password # test purposes only
-      - ZETACORED_IMPORT_GENESIS_DATA=${ZETACORED_IMPORT_GENESIS_DATA}
+      - ZETACORED_IMPORT_GENESIS_DATA=${ZETACORED_IMPORT_GENESIS_DATA-}
```
Comment (CryptoFewka) [inline contrib/localnet/docker-compose.yml:62]: Is this a typo or is a timespan missing?

**Review by CharlieMc0 — APPROVED** [review]

File: contrib/localnet/docker-compose.yml [hunk 2]
```diff
@@ -233,9 +238,9 @@ services:
         ipv4_address: 172.20.0.2
     entrypoint: ["/work/start-zetae2e.sh", "local"]
     environment:
-      - LOCALNET_MODE=${LOCALNET_MODE}
-      - E2E_ARGS=${E2E_ARGS}
-      - UPGRADE_HEIGHT=${UPGRADE_HEIGHT}
+      - LOCALNET_MODE=${LOCALNET_MODE-}
+      - E2E_ARGS=${E2E_ARGS-}
+      - UPGRADE_HEIGHT=${UPGRADE_HEIGHT-}
```
Comment (CryptoFewka) [inline contrib/localnet/docker-compose.yml:243]: Are these missing timespans?

**Review by CryptoFewka — COMMENTED** [review]
Looks like some timespans are missing, or there is an extra hyphen where there shouldn't be.

**Review by CryptoFewka — APPROVED** [review]
TIL about docker variable interpolation syntax. Looks good to me.

File: contrib/localnet/docker-compose.yml (hunk 2 above)
Comment (gartnera) [reply contrib/localnet/docker-compose.yml:243]: docker compose has some specific variable interpolation syntax: https://docs.docker.com/compose/environment-variables/variable-interpolation/#interpolation-syntax

so `${VAR-}` will expand to empty string if `VAR` is unset

**Pull request merged**